packages/core/src/fhir-normalization/resources/observation.ts

```diff
@@ -0,0 +1,129 @@
+import { Observation, Quantity, ObservationReferenceRange } from "@medplum/fhirtypes";
+import convert, { Unit } from "convert-units";
+
+type UnitComplex = {
+  unit: string;
+  code?: string;
+};
+/**
+ * This map is used to normalize unconventional unit names to standard unit names. i.e cel -> C, millimeter to m, etc.
+ */
+const unitNormalizationMap = new Map<string, string>([
+  ["cel", "C"],
+  ["degf", "F"],
+]);
+
+/**
+ * This map is used to convert standard units to preferred units. i.e C -> F, g -> kg, etc.
+ */
+const unitConversionMap = new Map<string, UnitComplex>([
+  ["C", { unit: "F", code: "degF" }],
+  ["F", { unit: "F", code: "degF" }],
+  ["g", { unit: "lb" }],
+  ["kg", { unit: "lb" }],
+  ["m", { unit: "in" }],
+  ["cm", { unit: "in" }],
+]);
+
+export function normalizeObservations(observations: Observation[]): Observation[] {
+  observations.map(o => {
+    if (o.valueQuantity) {
+      const initialValueUnit = processValueQuantity(o.valueQuantity);
+
+      if (o.referenceRange) {
+        processReferenceRanges(o.referenceRange, initialValueUnit);
+      }
+    }
+  });
+
+  return observations;
+}
+
+function processValueQuantity(valueQuantity: Quantity): string | undefined {
+  const result = getConvertedValueAndUnit(valueQuantity);
+  const initialUnit = valueQuantity.unit;
+  if (!result) return initialUnit;
+
+  const { newValue, newUnit, code } = result;
+
+  valueQuantity.value = newValue;
+  valueQuantity.unit = newUnit;
+  if (code) valueQuantity.code = code;
+
+  return initialUnit;
+}
```

Comment:
This goes against the conversations we had about immutability - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#10eb51bb9040806fa1deeeffd3c811ac).

A function should not update the inbound object and it should return the result of its processing, in this case, an instance of `Quantity`.

If you agree and take on updating this, please review other newly added functions as well.